### PR TITLE
[Backport] 8241825: Make compressed oops and compressed class pointer…

### PR DESCRIPTION
**Pull request by mmyxym**
…s independent(x86_64)

Summary: Make UseCompressedOops and UseCompressedClassPointers independent on x86_64

Test Plan: CICD

Reviewed-by: kuaiwei, yulei, lingjun

Issue: https://github.com/dragonwell-project/dragonwell11/issues/669

### CODE REVIEW

File: src/hotspot/cpu/ppc/globalDefinitions_ppc.hpp [hunk 1]
```diff
@@ -62,4 +62,6 @@ const bool CCallingConventionRequiresIntsAsLongs = true;
 // Define the condition to use this -XX flag.
 #define USE_POLL_BIT_ONLY UseSIGTRAP
 
+#define COMPRESSED_CLASS_POINTERS_DEPENDS_ON_COMPRESSED_OOPS true
```
Comment (yuleil) [inline src/hotspot/cpu/ppc/globalDefinitions_ppc.hpp:65]: 原patch 8241825 a/src/hotspot/cpu/ppc/globalDefinitions_ppc.hpp
```#define COMPRESSED_CLASS_POINTERS_DEPENDS_ON_COMPRESSED_OOPS false```
这里为啥改成true了

**Review by yuleil — APPROVED** [review]

**Comment by yuleil** [comment]
x64的build挂了，可以参考 https://github.com/dragonwell-project/dragonwell17/commit/7357e1d9f1b6aa2adb734b979e88c738c7f0be6b 修复下

File: src/hotspot/cpu/ppc/globalDefinitions_ppc.hpp (hunk 1 above)
Comment (mmyxym) [reply src/hotspot/cpu/ppc/globalDefinitions_ppc.hpp:65]: ppc我们没有测试，就让它保持原样了

**Review by D-D-H — APPROVED** [review]
LGTM

**Review by kuaiwei — APPROVED** [review]
LGTM

**Pull request merged**